src/mainwindow.cpp

```diff
@@ -193,9 +196,20 @@ void MainWindow::openImageFile(QString fileName) {
     fm_path_unref(path);
     return;
   }
-  // load the image file asynchronously
-  loadImage(path);
-  loadFolder(fm_path_get_parent(path));
+  Fm::FileInfo info = Fm::FileInfo::newFromNativeFile(nullptr,
+                                                      qPrintable(fileName),
+                                                      nullptr);
+  if (info.isDir()) {
```

Comment:
Thank you for your review @tsujan .
Yes, it is indeed simpler. Hence my questions in the first message. What is the policy regarding the use of Qt functions versus libfm-qt functions where both provide similar functionalities?